### PR TITLE
Express: GoogleCloud hosting option

### LINKED ISSUE #3298: Issue with "Express Tutorial Part 7: Deploying to production": (please mention Google Cloud Free Tier)
MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment

#### What information was incorrect, unhelpful, or incomplete?

>The good news when you're starting out is that there are quite a few sites that provide computing environments for "free", albeit with some conditions. For example, Heroku provides a free but resource-limited PaaS environment "forever", while Amazon Web Services and Microsoft Azure provide free credit when you first join.
>Many providers also have a "basic" tier that provides more useful levels of computing power and fewer limitations. Digital Ocean is an example of a popular hosting provider that offers a relatively inexpensive basic computing tier (in the $5 per month lower range at time of writing).

Everything is correct but let me ask to add a mention of Google Cloud as it'd be fair to mention all three big players.

#### Specific section or headline?

"Choosing a hosting provider"

#### What did you expect to see?

>[Google Cloud provides you both](https://cloud.google.com/free): Free Tier for free-of-change usage of some services (inc. popular for [site hosting](https://cloud.google.com/solutions/web-hosting) App Engine, Cloud Run and Cloud Storage) and credits.

right after "while Amazon Web Services and Microsoft Azure provide free credit when you first join" text.

#### Did you test this? If so, how?

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/deployment`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/deployment
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/deployment/index.html
* Last commit: https://github.com/mdn/content/commit/4b9d9a8bf3ab3db9f7b1efc89b5a91bbf459fe07
* Document last modified: 2021-03-01T12:15:15.000Z

</details>

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #3298

Google Cloud is another option/peer or azure, Amazon etc, so added that. 
Also fixed up note boxes to use standard notecard classes.

### CODE REVIEW

**Comment by CreaTorAlexander** [comment]
```
<div class="notecard note">
  <h4>Note</h4>

```
Just a question becuase I also saw it in another PR of you and I don't understand it. For what is this *div classname and heading update* above?

**Comment by hamishwillee** [comment]
@CreaTorAlexander This is the "proper" syntax for a note `div`- as documented [here](https://developer.mozilla.org/en-US/docs/MDN/Guidelines/CSS_style_guide#.note.notecard). There is no requirement to change these (or there would be a flaw), and arguably it isn't worth bothering because when we eventually move to markdown all of these will be fixed up in one go. 

I do it because I'm a little OCD :-) But only when I'm fixing up other things anyway.

**Comment by hamishwillee** [comment]
@Ryuno-Ki Could you please review this?

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html [hunk 1]
```diff
@@ -94,12 +97,13 @@ <h2 id="Choosing_a_hosting_provider">Choosing a hosting provider</h2>
...
 </ul>
 
-<p>The good news when you're starting out is that there are quite a few sites that provide computing environments for "free", albeit with some conditions. For example, <a href="https://www.heroku.com/">Heroku</a> provides a free but resource-limited <em>PaaS</em> environment "forever", while <a href="https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/billing-free-tier.html">Amazon Web Services</a> and <a href="https://azure.microsoft.com/en-us/pricing/details/app-service/">Microsoft Azure</a> provide free credit when you first join.</p>
+<p>The good news when you're starting out is that there are quite a few sites that provide computing environments for "free", albeit with some conditions. For example, <a href="https://www.heroku.com/">Heroku</a> provides a free but resource-limited <em>PaaS</em> environment "forever", while <a href="http://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/billing-free-tier.html">Amazon Web Services</a>, <a href="https://cloud.google.com/free/docs/gcp-free-tier">Google Cloud</a>, and <a href="https://azure.microsoft.com/en-us/pricing/details/app-service/">Microsoft Azure</a> provide free credit when you first join.</p>
```
Comment (Ryuno-Ki) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.html:100]: Does Amazon not offer https?

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html [hunk 2]
```diff
@@ -154,8 +160,9 @@ <h3 id="Log_appropriately">Log appropriately</h3>
 export DEBUG="author,book"
 </pre>
 
-<div class="note">
-<p><strong>Challenge:</strong> Calls to <code>debug</code> can replace logging you might previously have done using <code>console.log()</code> or <code>console.error()</code>. Replace any <code>console.log()</code> calls in your code with logging via the <a href="https://www.npmjs.com/package/debug">debug </a>module. Turn the logging on and off in your development environment by setting the DEBUG variable and observe the impact this has on logging.</p>
+<div class="notecard note">
+  <h4>Challenge</h4>
+  <p>Calls to <code>debug</code> can replace logging you might previously have done using <code>console.log()</code> or <code>console.error()</code>. Replace any <code>console.log()</code> calls in your code with logging via the <a href="https://www.npmjs.com/package/debug">debug </a>module. Turn the logging on and off in your development environment by setting the DEBUG variable and observe the impact this has on logging.</p>
```
Comment (Ryuno-Ki) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.html:165]:
```suggestion
  <p>Calls to <code>debug</code> can replace logging you might previously have done using <code>console.log()</code> or <code>console.error()</code>. Replace any <code>console.log()</code> calls in your code with logging via the <a href="https://www.npmjs.com/package/debug">debug</a> module. Turn the logging on and off in your development environment by setting the DEBUG variable and observe the impact this has on logging.</p>
```
Sorry, my OCD over here :sweat_smile:

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html [hunk 3]
```diff
@@ -189,8 +196,9 @@ <h3 id="Use_gzipdeflate_compression_for_responses">Use gzip/deflate compression
 ...
 </pre>
 
-<div class="note">
-<p><strong>Note</strong>: For a high-traffic website in production you wouldn't use this middleware. Instead, you would use a reverse proxy like <em>Nginx</em>.</p>
+<div class="notecard note">
+  <h4>Note</h4>
+  <p>For a high-traffic website in production you wouldn't use this middleware. Instead, you would use a reverse proxy like <em>Nginx</em>.</p>
```
Comment (Ryuno-Ki) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.html:201]:
```suggestion
  <p>For a high-traffic website in production you wouldn't use this middleware. Instead, you would use a reverse proxy like <em>nginx</em>.</p>
```
Looking through https://nginx.org/en/docs/faq.html it appears, that `nginx` is the product and `Nginx` is the company (but the literature is inconsistent here).

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html [hunk 4]
```diff
@@ -213,8 +221,9 @@ <h3 id="Use_Helmet_to_protect_against_well_known_vulnerabilities">Use Helmet to
 <strong>app.use(helmet())</strong>;
 ...</pre>
 
-<div class="note">
-<p><strong>Note:</strong> The command above adds a <em>subset</em> of the available headers. They make sense for most sites. You can add/disable specific headers as needed by following the instructions on <a href="https://www.npmjs.com/package/helmet">npm</a>.</p>
+<div class="notecard note">
+  <h4>Note</h4>
+  <p>The command above adds a <em>subset</em> of the available headers. They make sense for most sites. You can add/disable specific headers as needed by following the instructions on <a href="https://www.npmjs.com/package/helmet">npm</a>.</p>
```
Comment (Ryuno-Ki) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.html:226]: The link text is a bit confusing here. Perhaps it should become
```suggestion
  <p>The command above adds a <em>subset</em> of the available headers. They make sense for most sites. You can add/disable specific headers as needed by following the instructions on <a href="https://www.npmjs.com/package/helmet">helmet</a>.</p>
```
?

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html [hunk 5]
```diff
@@ -322,17 +332,19 @@ <h3 id="Creating_an_application_repository_in_Github">Creating an application re
 </ol>
 
 <div class="notecard warning">
-<p><strong>Warning:</strong> In 2020 Github changed the default repo branch name to "main" (from "master"). If using an older/existing repository you might need to call <code>git push origin master</code> instead.</p>
+  <h4>Warning</h4>
+  <p>In 2020 Github changed the default repo branch name to "main" (from "master"). If using an older/existing repository you might need to call <code>git push origin master</code> instead.</p>
```
Comment (Ryuno-Ki) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.html:336]: I believe, they stylise their product as
```suggestion
  <p>In 2020 GitHub changed the default repo branch name to "main" (from "master"). If using an older/existing repository you might need to call <code>git push origin master</code> instead.</p>
```

**Review by Ryuno-Ki — CHANGES_REQUESTED** [review]
So far, looks good to me.
Please, address the Amazon and the broken link and this can be merged.

Bonus points, if you would update the product names I called out :-)

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/learn/server-side/express_nodejs/deployment/index.html:100]: No idea how that crept in.

File: files/en-us/learn/server-side/express_nodejs/deployment/index.html (hunk 4 above)
Comment (hamishwillee) [reply files/en-us/learn/server-side/express_nodejs/deployment/index.html:226]: I modified slightly differently. Thanks.

**Comment by hamishwillee** [comment]
Thanks @Ryuno-Ki - fixed!

**Review by Ryuno-Ki — APPROVED** [review]
Looks great!

**Comment by hamishwillee** [comment]
Thanks for reviewing @Ryuno-Ki !

**Pull request merged**